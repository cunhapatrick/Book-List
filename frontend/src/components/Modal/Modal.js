import React from 'react';
import PropTypes from 'prop-types';

// CSS
import  {
  Wrap,
} from './Modal.style';

const Modal = ({ show, data, closeModal }) => (
  <Wrap onClick={closeModal} className={show ? 'show' : ''}>
    <div>

      <div>
        <h3>Titulo do livro</h3>
        <p>{data.title}</p>
      </div>
      <div>
        <h3>Autor da obra</h3>
        <p>{data.author}</p>
      </div>
      <div>
        <h3>Editora Responsavel</h3>
        <p>{data.publisher}</p>
      </div>
      <div>
        <h3>Ano de Publicação</h3>
        <p>{data.year}</p>
      </div>
      <div>
        <h3>Idioma</h3>
        <p>{data.language}</p>
      </div>
      <div>
        <h3>Peso em g</h3>
        <p>{data.weight} gramas</p>
      </div>
      <div>
        <h3>Dimensões</h3>
        <p>comprimento: {data.dimensions[0]}cm</p>
        <p>altura: {data.dimensions[1]}cm</p>
        <p>largura: {data.dimensions[2]}cm</p>
      </div>
      <button onClick={closeModal}>Fechar</button>
    </div>
  </Wrap>
);

Modal.propTypes = {
  closeModal: PropTypes.func.isRequired,
  show: PropTypes.bool.isRequired,
  data: PropTypes.object.isRequired,
}

export default Modal;