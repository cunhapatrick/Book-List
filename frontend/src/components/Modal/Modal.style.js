import styled from 'styled-components';

export const Wrap = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: rgba(0,0,0,0.2);
  z-index: 99999;
  opacity:0;
  pointer-events: none;
  text-align:center;

  &.show {
    opacity:1;
    pointer-events: auto;
  }
  > div {
    width: 400px;
    height: 800px;
    position: relative;
    margin: 10% auto;
    background: #fff;

    div {
      text-align: left;
      h3, p { padding: 10px; }
      p { padding-left: 15px; }
      &:last-of-type {
        width: 40%;
        margin: 0;
      }
    }

    button {
      border-radius: 20px;
      padding: 10px;
      font-weight: bold;
    }
  }
`;