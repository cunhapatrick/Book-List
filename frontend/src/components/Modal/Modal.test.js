import React from 'react';
import { shallow } from 'enzyme';
import Modal from './Modal';

describe('Testing Modal Component', () => {
  it('should render properly', () => {
    const wrapper = shallow(
      <Modal
        closeModal={() => {}}
        show={true}
        data={{
          title: 'teste',
          dimensions: [10, 5, 8]
        }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});