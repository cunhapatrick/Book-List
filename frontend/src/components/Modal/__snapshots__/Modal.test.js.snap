// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Modal Component should render properly 1`] = `ShallowWrapper {}`;
